### ISSUE TITLE
Group argument in validation is null when calling 'setRules'

### ISSUE THREAD

**Issue by maciejkaiser** [mistic100/jQuery-QueryBuilder#701]
Hi,

I have an issue, after update to latest version of this plugin, with setting rules.
After declare and fill with filters is ok, but the problem is when I just call :
`$('#ruleGenerator1').queryBuilder('setRules', {"condition":"AND","rules":[{"id":"8","field":"8","operator":"greater","value":["1"],"not":false}]});`

then in console 

![querybuilder](https://user-images.githubusercontent.com/12011503/40607985-59130000-626a-11e8-914b-f7f9b3796460.PNG)

Any ideas what I'm doing wrong or how to fix it?

**Comment by mistic100** [comment]
Don't you have a listener tha call "getRules" on the "rulesChanged" event ?
You must disable it when calling "setRules".

**Comment by maciejkaiser** [comment]
Thanks! It works :) But I need to watch changes and save it into variable... Is there any possibility to do that? Maybe setTimeout?

**Comment by mistic100** [comment]
No, just set a flag somewhere before calling setRules, and use this flag in the event handler to do nothing.

**Comment by mistic100** [comment]
I mark this issue has bug to make sure the "getRules" method does not crash when the builder is totally empty.

**Comment by yanielbf** [comment]
Is this issue fixed??? I have some trouble. Help me please!!!!. Greeting.
